src/Microsoft.SqlTools.ServiceLayer/QueryExecution/SpecialAction.cs

```diff
@@ -0,0 +1,89 @@
+// 
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+
+namespace Microsoft.SqlTools.ServiceLayer.QueryExecution
+{
+    /// <summary>
+    /// Class that represents a Special Action which occured by user request during the query 
+    /// </summary>
+    public class SpecialAction {
+        
+        #region Private Class variables 
+
+        // Underlying representation as bitwise flags to simplify logic
+        [Flags]
+        private enum ActionFlags 
+        {
+            None    = 0,
+            // All added options must be powers of 2
+            ExpectYukonXmlShowPlan = 1
+        }
+
+        private ActionFlags flags; 
+        private bool none;
+        private bool expectYukonXmlShowPlan;
+
+        #endregion
+
+        /// <summary>
+        /// The type of XML execution plan that is contained with in a result set  
+        /// </summary>
+        public SpecialAction()
+        {
+            flags = ActionFlags.None;
+        }
+
+        #region Public Functions
+        /// <summary>
+        /// No Special action performed 
+        /// </summary>
+        public bool None 
```

Comment:
So, I think you don't need setters for these properties. Additionally, you can avoid storing booleans for none, etc by just returning `flags == ActionFlags.None`. For the others, you can just do `flags & ActionFlags.Yukon > 0`

If you're adding the setters because you need to be able to deserialize a SpecialActions class from json, you should explicitly check the value provided. If it's true, `|=` the value to the flags, if it's false, `&= ~(value)` to the flags.